Formula/sapling.rb

```diff
@@ -0,0 +1,33 @@
+class Sapling < Formula
+  desc "Source control client"
+  homepage "https://sapling-scm.com"
+  url "https://github.com/facebook/sapling/archive/refs/tags/0.1.20221118-210929-cfbb68aa.tar.gz"
+  version "0.1.20221118-210929-cfbb68aa"
+  sha256 "51ce336f1eb382e591a7384d0292cb41fafbbd92cc701ee74b50099beece65e4"
+  license "GPL-2.0-or-later"
+  head "https://github.com/facebook/sapling.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "rust" => :build
+  depends_on "yarn" => :build
+  depends_on "gh"
+  depends_on "node"
+  depends_on "openssl@1.1"
+  depends_on "python@3.8"
```

Comment:
Does it not work with OpenSSL 3 and python 3.11?

Reply:
It actually doesn't work with Python 3.11. We are currently [investigating that issue](https://github.com/facebook/sapling/issues/144).

We haven't tested if it actually works with a different OpenSSL version, but I can take a look.

Reply:
It seems that we cannot use OpenSSL 3 after all, since Python 3.8 depends on OpenSSL 1.1, and this seems to make `brew audit --new sapling` fail:

```
sapling:
  * sapling contains conflicting version recursive dependencies:
      openssl@1.1, openssl@3
    View these with `brew deps --tree sapling`.
Error: 1 problem in 1 formula detected`
```

Reply:
Yeah, python 3.8 is far too old to let new formula depend on either way. It'll need a newer python.

Reply:
Ok, just updated it to Python 3.10

Reply:
Well, Python 3.9 and Python 3.10 don't work either for different reasons. Our build fails with:

```
running install_lib
creating /Users/sggutier/homebrew/lib/python3.9/site-packages/ghstack
error: could not create '/Users/sggutier/homebrew/lib/python3.9/site-packages/ghstack': Operation not permitted
make: *** [install-oss] Error 1
```

Are there any guidelines that explain why Python 3.8 is too old? It seems to be supported until October 2024:
https://peps.python.org/pep-0569/

Reply:
https://docs.brew.sh/Acceptable-Formulae#stuff-that-doesnt-build-with-the-latest-stable-xcode-clang is a little more restricted but we should really update it to mean "no old dependencies". If upstream can't fix python compatibility in 3 years, I'm not confident they will in another 2 leaving Homebrew with a difficult to maintain package with unsupported dependencies.

Reply:
For a number of reasons, at our company [we use Python 3.8](https://github.com/facebookincubator/cinder). It's currently being upgraded to 3.10, and we weren't really that concerned with being compatible with the latest Python versions until we started seeing issues with the open source version.

In any case, we are expecting to fix our compatibility issues with Python at some point in the next two weeks.

Reply:
We made a few changes so that the Homebrew formula would be able to build with Python 3.9-3.11 . We are still, however, trying to make sure that updating to 3.9+ on macOS arm doesn't introduce some subtle breakages (at the moment it still hasn't).

Reply:
> Well, Python 3.9 and Python 3.10 don't work either for different reasons. Our build fails with:
> 
> ```
> running install_lib
> creating /Users/sggutier/homebrew/lib/python3.9/site-packages/ghstack
> error: could not create '/Users/sggutier/homebrew/lib/python3.9/site-packages/ghstack': Operation not permitted
> make: *** [install-oss] Error 1
> ```

This is trying to install the `ghstack` module outside of the formula's prefix. If your `Makefile` invokes a `setup.py` (or similar), this is usually fixed by specifying the correct destination for `--install-lib`. It should be something like `prefix/Language::Python.site_packages("python3.x", prefix)` (using the `prefix` and `site_packages` methods defined by `brew`).

Reply:
Sounds good to me. Actually just yesterday I sent a commit for bumping our [GitHub actions bottle to 3.11](https://github.com/facebook/sapling/commit/81685da429b7245f939792506d2b3568db1a19fe).

That commit explains what changes we had to make in order to make to compile and run. We'll test this a bit more just to be sure that there are no further issues with Python 3.11